### DISCUSSION TITLE
Additional toolbar buttons

### DISCUSSION THREAD

**Discussion by vertigo220** [doublecmd/doublecmd#1972] [Suggestions]
Add new toolbar button options:

- Delete permanently (bypass recycle bin)
- Create files of different types (e.g. text, shortcut)
- Activate a hot key, which would allow for doing other functions not explicitly provided (this would, for example, allow for the permanently delete function since that can be assigned to a hotkey, which could then be assigned to a button)

**Comment by alexx2000** [comment]
> Delete permanently (bypass recycle bin)

You can add a such button, see [cm_Delete ](https://doublecmd.github.io/doc/en/cmds.html#cm_Delete)command.

> Create files of different types (e.g. text, shortcut)

You can add a such buttons. I don't know which file types you need.

> Activate a hot key, which would allow for doing other functions not explicitly provided (this would, for example, allow for the permanently delete function since that can be assigned to a hotkey, which could then be assigned to a button)

You can [assign](https://doublecmd.github.io/doc/en/configuration.html#ConfigHotKeys) any hotkey to any command. Default hotkeys for the permanently delete are `Shift+Delete` and `Shift+F8`.

**Reply by vertigo220** [reply]
> You can add a such button, see [cm_Delete ](https://doublecmd.github.io/doc/en/cmds.html#cm_Delete)command.

Ok, didn't realize you could use parameters in that way. That makes sense and is actually a good way of doing it, though it would be helpful if the available parameters were selectable there, both to make it more obvious and to help in setting it up. Otherwise, the user has to refer to the cmds.htm link _and_ make sense of it. Since there are no examples, I wasn't sure how to format it, so I had to refer to the hotkeys setup to figure that out (to see the formatting is trashcan=...), then determine that even though in hotkeys it's set to "trashcan=reversesetting" I want to set it specifically to "trashcan=0" to ensure it always does what I want. Now that I understand how it all works, I can see it's very powerful and allows for a lot of customization, but the process could definitely be made easier.

> You can add a such buttons. I don't know which file types you need.

How, though? I looked under Make..., Create..., and New... and don't see any option to create a file of any type, including a generic where the file type would be specified via parameters. There's cm_EditNew, but that's only for text files (yet doesn't append .txt to them) and opens them for editing, when I just want to create a new text file that will be selected so I can immediately hit F2 and rename it. Perhaps I'm missing something, but then that would just speak to the fact it's not clear enough how to do this particular thing.

> You can [assign](https://doublecmd.github.io/doc/en/configuration.html#ConfigHotKeys) any hotkey to any command. Default hotkeys for the permanently delete are `Shift+Delete` and `Shift+F8`.

I know this. I was saying sort of the opposite. Instead of assigning a hotkey to do a function, it should allow you to create a button that, when clicked, would be the same as hitting the hotkey. So if I want to have a button do something, but there's no way to create a button to do that, I could assign a hotkey to do it then create a button that activates the hotkey, thereby performing its assigned function.

**Reply by alexx2000** [reply]
> How, though? 

DC itself can create directly only text files. There are various possible variants here, see [this discussion](https://doublecmd.h1n.ru/viewtopic.php?p=25326) as example. Also you can use the [New](https://doublecmd.h1n.ru/viewtopic.php?p=36647#p36647) sub-menu.

> I know this. I was saying sort of the opposite.

Button execute a command. This is how the buttons work in the DC. I don't plan to change this.

**Reply by Skif-off** [reply]
> Since there are no examples...

[Without parameters and with parameters](https://doublecmd.github.io/doc/en/toolbar.html#catinternal).

**Reply by vertigo220** [reply]
Thanks. I meant no examples on the link provided. As I mentioned, it's preferable, when possible and reasonable, to just make things easier without having to refer to help, and in this case, it involves finding two separate places in help, one with the parameters, and one explaining their usage. Just a lot of digging to do something that should require little to none. I'll admit, I didn't do my due diligence on this, including simply clicking the "Help" link. Part of that is that there's just a lot going on, so it's a bit overwhelming at first, and part is that the options page is pretty busy and I don't think I even saw that, and part is that I generally like to get a feel for how intuitive a software is and try to use it by what's obvious vs digging into help so I can offer that perspective, as it can be useful for providing a fresh perspective (it can be difficult to see these things when you're the developer or you've been using it for a while). So I'm just throwing it out there in case it helps tweak things to make it easier to use.

**Reply by Skif-off** [reply]
> I generally like to get a feel for how intuitive a software is and try to use it by what's obvious vs digging into help

A simple example. When the capabilities of the intuitive Microsoft Paint become insufficient for you, you turn your attention to more advanced applications - Paint.NET, GIMP, Krita, PhotoShop, Inkscape, Blender and so on... I am not ready to believe that you will not open help files, Dan Margulis's "The Classic Guide to Color Correction" and other books or articles with good examples, a review on Youtube, internet search :) (I mean any user, not specifically you, it just makes it easier to build a sentence.)
And I think among the programs that you use regularly were those that took time and effort, even taking into account existing experience. Like all of us :))

I'm not against simpler interaction, but I understand that simplification always has limits. And no one threw us out of the boat to teach us how to swim :) For example, the "Help" button and the F1 hotkey are available in each settings section, there is a link to the description of this command in the command choose dialog.

P.S. I have updated the [screenshot](https://skif-off.github.io/doublecmd.github.io/doc/en/images/imgDC/pic33.png), maybe along with the description, this will make the usage more understandable.

**Comment by vertigo220** [comment]
Well, I just typed a whole response which was then lost when I clicked that link (thanks, GitHub). So in short, yes, I refer to help, often, but don't always have the time or the will, considering how much time I can sometimes end up spending doing so. So when something can be changed to reduce that need, that's always good, within reason of course.
